analysis/src/CreateInterface.ml

```diff
@@ -250,11 +254,22 @@ let printSignature ~extractor ~signature =
       Buffer.add_string buf (indent ^ newItemStr ^ "\n");
       processSignature ~indent items
     | Sig_type (id, typeDecl, resStatus) :: items ->
-      let newItemStr =
-        sigItemToString
-          (Printtyp.tree_of_type_declaration id typeDecl resStatus)
+      let lines =
+        let posStart, posEnd = Loc.range typeDecl.type_loc in
+        extractor |> SourceFileExtractor.extract ~posStart ~posEnd
       in
-      Buffer.add_string buf (indent ^ newItemStr ^ "\n");
+      let attributes = AttributesUtils.make lines in
+
+      (if not (AttributesUtils.isEmpty attributes) then
+         (* Copy the type declaration verbatim to preserve attributes *)
```

Comment:
Does this work in the `else` branch too? Can we always copy verbatim? Or what's the tradeoff here?